# 错误笔记

## Class 实操错误笔记

- （1） class 类内部某个功能调用，无需带入 `self 这个形参`，带入了就会报错，请切记；

```python
    # 以下就是错误的带入了 `self` 参数后报错了；
    def crawl_last_day_corona_virus(self, tag_id):
        # 1.发送源码下载请求，并返回源码
        home_page = self.get_content_from_url(self.home_url)
        # 2. 提取引起字符串数据，并转为python对象
        data = self.parse_home_page(self, home_page, tag_id)
        # 3. 保证python数据为json文件
        self.save_to_json(self, r"data/last_day_corona_virus.json", data)
```



## bs4 数据提取错误笔记

- （1）标签提取中 id = "xxx" 这里的id是固定形参，不能随便调整；

```python
# 错误的 id 标签提取
def parse_home_page(self, home_page, tag_id):
	soup = BeautifulSoup(home_page, 'lxml')
	script_tag = soup.find(tag_id)
	script_string = script_tag.string

# 正确的 id 标签提取应该如下
def parse_home_page(self, home_page, tag_id):
	soup = BeautifulSoup(home_page, 'lxml')
	script_tag = soup.find(id=tag_id)
	script_string = script_tag.string
```

## 列表添加方法错误笔记

- （1）使用列表的添加方法向其中添加数据，无趣在重新赋值

```python
# 错误的重新赋值
corona_virus = []
for x in range(10):
    corona_virus = corona_viurs.append(x)
    
# 正确的方法是直接使用即可
corona_virus = []
for x in range(10):
    corona_virus.append(x)
```

